.github/workflows/preview.yml

```
@@ -0,0 +1,42 @@
+name: Preview
+on:
+  pull_request:
+    branches: 
+      - master
+      - develop
+      - 'release/**'
+jobs:
+  deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      
+      - name: install deps
+        run: yarn
+
+      - name: build docs
+        run: yarn workspace @okta/design-docs build
+
+      - name: deploying docs to ods.dev
+        run: | 
+          SHA=$(git rev-parse --short ${{ github.event.pull_request.head.sha || github.sha }})
+          aws s3 sync ./packages/docs/dist/ s3://ods.dev/$SHA --delete
```

nit: Let's add a version indicator to our bucket:

```suggestion
          aws s3 sync ./packages/docs/dist/ s3://ods.dev/$DOCS_VERSION/$SHA --delete
```

---

I spoke to @jmelberg-okta offline about this. My assumption was that this would be used to version docs, the idea here is to instead to allow for an easy way to clean up the bucket and remove any old cruft/builds.

I think a better solution is object expiration, this way it happens and we don't have to do any cleanup manually. What those rules are is an interesting question. Perhaps an object should expire after 30 days? 90 days?

I don't have any strong preference here. Let me know what you think @jmelberg-okta @edburyenegren-okta

---

Gut: 30 days is probably safe. If someone has not pushed an update to a PR in 30 days, that PR is probably dead.

However, there are some valid reasons that might be the case - maybe someone spins up a PR to prove something is an issue with a POC solve, but we won't get to it for a full sprint.

If 60/90 days doesn't introduce any serious storage issues (our `dist` is tiny, so this feels safe), that feels better to me.

---

I have had PRs that have gone longer than 30 days, those were rare exceptions though - and frankly, were as a result of something that went wrong (process-wise). I don't see any harm in keeping the preview for a while, 90 days or even longer.

---

Let's settle on 30 days for now. If needed, we always can push an empty commit to the topic-branch to kick-off a new docs deploy.

---

great point, not sure why I didnt think of it 😂